drivers/modem/hl7800.c

```diff
@@ -5182,14 +5255,10 @@ static int modem_reset_and_configure(void)
 				bands_bottom, ictx.mdm_bands_bottom);
 		}
 
-		snprintk(new_bands, sizeof(new_bands),
-			 "AT+KBNDCFG=%d,%04x%08x%08x", ictx.mdm_rat, bands_top,
-			 bands_middle, bands_bottom);
-		SEND_AT_CMD_EXPECT_OK(new_bands);
-
-		SEND_AT_CMD_EXPECT_OK("AT+CFUN=1,1");
+		snprintk(new_bands, sizeof(new_bands), "%04x%08x%08x", bands_top, bands_middle,
```

Comment:
You could do it like this so it follows the macros

```suggestion
		snprintk(new_bands, sizeof(new_bands), "%0"STRINGIFY(MDM_TOP_BAND_SIZE)"x%0"STRINGIFY(MDM_MIDDLE_BAND_SIZE)"x%0"STRINGIFY(MDM_BOTTOM_BAND_SIZE)"x", bands_top, bands_middle,
```